tests/js/searchtools.js

```diff
@@ -31,6 +31,38 @@ describe('Basic html theme search', function() {
 
 });
 
+describe("htmlToText", function() {
+
+  const testHTML = `<html>
+  <div class="body" role="main">
+    <section id="getting-started">
+      <h1>Getting Started</h1>
+      <p>Some text</p>
+    </section>
+    <section id="other-section">
+      <h1>Other Section</h1>
+      <p>Other text</p>
+    </section>
+    <section id="yet-another-section">
+      <h1>Yet Another Section</h1>
+      <p>More text</p>
+    </section>
+  </div>
+  </html>`;
+
+  it("basic case", () => {
+    expect(Search.htmlToText(testHTML).trim().split(/\s+/)).toEqual([
+      'Getting', 'Started', 'Some', 'text', 
+      'Other', 'Section', 'Other', 'text', 
+      'Yet', 'Another', 'Section', 'More', 'text'
+    ]);
+  });
+
+  it("will start reading from the anchor", () => {
+    expect(Search.htmlToText(testHTML, '#other-section').trim().split(/\s+/)).toEqual(['Other', 'Section', 'Other', 'text']);
```

Comment:
Is the `trim...split` behaviour here to avoid being too-precise about whitespace in the output?

If so, an alternative could be to make a less-restrictive assertion; something like `retrievedText.startsWith("Other Section")`.

Reply:
That's right, just trying not to test whitespace. I think I prefer to check for all the tokens in the output, feels more comprehensive to me.